### ISSUE TITLE
Perimeter feature is incorrect

### ISSUE THREAD

**Issue by Nicholas-Schaub** [PolusAI/nyxus#69]
When getting area and perimeter features for the demo image `0acd2c223d300ea55d0546797713851e818e5c697d073b7f4091b96ce0f3d2fe.tif`, the following results are given.

```
                                          mask_image                                    intensity_image  label  AREA_PIXELS_COUNT  PERIMETER
0  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      1              134.0      169.0
1  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      2              266.0      324.0
2  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      3              192.0      272.0
3  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      4              256.0      330.0
4  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      5              279.0      360.0
5  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      6              392.0      506.0
6  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      7              267.0      324.0
7  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      8               21.0       25.0
8  0acd2c223d300ea55d0546797713851e818e5c697d073b...  0acd2c223d300ea55d0546797713851e818e5c697d073b...      9               95.0      117.0
```

At most, I would expect the perimeter to have an upper bound on area_pixel_count, but it is always higher.

It appears as though the error is caused by commented code here:
https://github.com/PolusAI/nyxus/blob/b066182b157616b3b295fdf0da7e33642e97cb8f/src/nyx/features/contour.cpp#L173

It looks like all pixels in the bounding box of the contour are being stored in the contour. I suspect this means the rest of the contour features are also incorrect.

**Comment by Nicholas-Schaub** [comment]
As a followup, it looks like the perimeter is calculated as a simple pixel count of border pixels. However, in a study that evaluated perimeter of different objects, it was found that Matlab was the most accurate and Matlab was the only algorithm that did not do a simple pixel count.

Has there been a comparison between Matlab perimeter calculations and Nyxus perimeter?

**Comment by Nicholas-Schaub** [comment]
I have been further looking into this, and it appears as though there may be an additional bug in the contour code.

Here are objects 20 (upper left) and 25 (lower right) from image `3b3f516ebc9a16cff287a5ffd3a1861a345a6d38bedbba74f1c0b0e0eac62afd.tif`
![image](https://user-images.githubusercontent.com/15925882/201488387-161715f9-c49f-43b5-8656-cc9a81c49308.png)

It appears as though the perimeter calculation counts pixels around the single background pixel in object 25, but not object 20. Based on my reading of the code, could this be due to the neighborhood approach that the code uses to identify border pixels?

**Comment by hsidky** [comment]
Fixed in #77
